pkg/result/processors/exclusion_generated_file_matcher_test.go

```diff
@@ -0,0 +1,154 @@
+		`// Generated by running
+//  maketables --tables=all --data=http://www.unicode.org/Public/8.0.0/ucd/UnicodeData.txt
+// --casefolding=http://www.unicode.org/Public/8.0.0/ucd/CaseFolding.txt
+// DO NOT EDIT`,
+		`/*
+* CODE GENERATED AUTOMATICALLY WITH github.com/ernesto-jimenez/gogen/unmarshalmap
+* THIS FILE SHOULD NOT BE EDITED BY HAND
+  */`,
+		`// AUTOGENERATED FILE: easyjson file.go`,
+		` * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)`,
+	}
+
+	for _, comment := range comments {
+		t.Run(comment, func(t *testing.T) {
+			t.Parallel()
+
+			generated := p.isGeneratedFileLax(comment)
+			assert.True(t, generated)
+		})
+	}
+}
+
+func TestGeneratedFileMatcher_isGeneratedFileLax_nonGenerated(t *testing.T) {
+	p := NewGeneratedFileMatcher(config.GeneratedModeLax)
+
+	comments := []string{
+		"code not generated by",
+		"test",
+	}
+
+	for _, comment := range comments {
+		t.Run(comment, func(t *testing.T) {
+			t.Parallel()
+
+			generated := p.isGeneratedFileLax(comment)
+			assert.False(t, generated)
+		})
+	}
+}
+
+func Test_isGeneratedFileStrict(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		filepath string
+		assert   assert.BoolAssertionFunc
+	}{
+		{
+			desc:     "go strict",
+			filepath: filepath.FromSlash("testdata/exclusion_generated_file_filter/go_strict.go"),
+			assert:   assert.True,
+		},
+		{
+			desc:     "go strict invalid",
+			filepath: filepath.FromSlash("testdata/exclusion_generated_file_filter/go_strict_invalid.go"),
+			assert:   assert.False,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			t.Parallel()
+
+			file, err := parser.ParseFile(token.NewFileSet(), test.filepath, nil, parser.PackageClauseOnly|parser.ParseComments)
+			require.NoError(t, err)
+
+			generated := isGeneratedFileStrict(file)
+
+			test.assert(t, generated)
+		})
+	}
+}
+
+func Test_getComments(t *testing.T) {
+	testCases := []struct {
+		fpath string
+		doc   string
+	}{
+		{
+			fpath: filepath.FromSlash("testdata/exclusion_generated_file_filter/exclude.go"),
+			doc: `first line
+second line
+third line
+this text also
+and this text also`,
+		},
+		{
+			fpath: filepath.FromSlash("testdata/exclusion_generated_file_filter/exclude_doc.go"),
+			doc:   `DO NOT EDIT`,
+		},
+		{
+			fpath: filepath.FromSlash("testdata/exclusion_generated_file_filter/exclude_block_comment.go"),
+			doc: `* first line
+ *
+ * second line
+ * third line
+and this text also
+this type of block comment also
+this one line comment also`,
+		},
+	}
+
+	for _, tc := range testCases {
+		file, err := parser.ParseFile(token.NewFileSet(), tc.fpath, nil, parser.PackageClauseOnly|parser.ParseComments)
+		require.NoError(t, err)
+
+		doc := getComments(file)
+		assert.Equal(t, tc.doc, doc)
+	}
+}
+
+// Issue 954: Some lines can be very long, e.g. auto-generated
+// embedded resources. Reported on file of 86.2KB.
```

Comment:
nit:
```suggestion
// Issue #954: Some lines can be very long, e.g. auto-generated embedded resources.
// Reported on file of 86.2KB.
```

When GitHub sees `#954` (or `golangci/golangci-lint#954`), it renders it as a link to the issue `https://github.com/golangci/golangci-lint/issues/954`

<img width="1134" alt="image" src="https://github.com/user-attachments/assets/7dad8c04-9fd5-4937-bdd9-e777a05dd5b5" />

This issue-link rendering from `#<num>` also works in popular IDEs.

Reply:
Oh, maybe it's only VS Code renders them.

<img width="1319" alt="Screenshot 2025-02-14 at 14 11 51" src="https://github.com/user-attachments/assets/400d81f5-77b6-49bb-9d49-b3e1d41a1994" />

Can be resolved.